submissions/nadsatt/friends-app/modules/filter-items.js

```diff
@@ -0,0 +1,173 @@
+import svgs from './svgs.js';
+
+class Filter {
+    constructor(optionsHTML, property, category, userService){
+        this.element = document.createElement('li');
+        this.element.innerHTML =
+            `<div class="filter-item__options">
+                ${optionsHTML}
+                <span class="filter-item__option" data-state="none">${svgs.none}</span>
+            </div>`;
+        this.element.classList.add('filter-item', `${category}-filter-item`);
+
+        this.property = property;
+        this.category = category;
+        this.name = `${property}-${category}`;
+        this.userService = userService;
+        this.defaultState = 'none';
+        this.defaultOptionElement = this.element.querySelector(`[data-state="${this.defaultState}"]`);
+        this.settedState = this.defaultState;
+        this.activatedOptionElement = this.defaultOptionElement;
+        this.classNames = {
+            openedFilterElement: 'filter-item--opened',
+            activatedOptionElement: 'filter-item__option--activated'
+        };
+    }
+
+    open(){
+        this.element.classList.add(this.classNames.openedFilterElement);
+    }
+
+    close(){
+        this.element.classList.remove(this.classNames.openedFilterElement);
+    }
+
+    setState(optionElement){
+        this.deactivateOptionAndIcon();
+        this.updateOptionAndState(optionElement);
+        if(this.settedState !== this.defaultState) this.activateOptionAndIcon();
+    }
+
+    applyAccordingToState(){
+        if(this.settedState !== this.defaultState){
+            this[`applyAccordingTo${this.getCapitalizedState(this.settedState)}State`]();
+        }
+    }
+
+    deactivateOptionAndIcon(){
+        this.activatedOptionElement.classList.remove(this.classNames.activatedOptionElement);
+        this.icon.deactivate();
+    }
+
+    activateOptionAndIcon(){
+        this.activatedOptionElement.classList.add(this.classNames.activatedOptionElement);
+        this.icon.activate();
+    }
+
+    updateOptionAndState(optionElement){
+        this.activatedOptionElement = optionElement;
+        this.settedState = optionElement.dataset.state;
+    }
+
+    getCapitalizedState(state){
+        return state[0].toUpperCase() + state.slice(1);
+    }
+}
+
+class SearchFilter extends Filter {
+    constructor(...args){
+        const optionsHTML =
+            `<input class="filter-item__option" type="text" data-state="search">`;
+        super(optionsHTML, ...args);
+
+        this.input = this.element.querySelector('input');
+    }
+
+    setState(optionElement){
+        this.deactivateOptionAndIcon();
+        this.updateOptionAndState(optionElement);
+
+        if(this.settedState === this.defaultState) this.input.value = ''
+        else if(this.input.value) this.activateOptionAndIcon();
```

Comment:
JS has brackets - `{}`. Please use them, they make code more readable :)

Additional line not a problem in 99,9% cases